web_src/js/features/eventsource.sharedworker.js

```diff
@@ -0,0 +1,149 @@
+self.name = 'eventsource.sharedworker.js';
+
+const sourcesByUrl = {};
+const sourcesByPort = {};
+
+class Source {
+  constructor(url) {
+    this.url = url;
+    this.eventSource = new EventSource(url);
+    this.listening = {};
+    this.clients = [];
+    this.listen('open');
+    this.listen('logout');
+    this.listen('notification-count');
+    this.listen('error');
+  }
+
+  register(port) {
+    const portIdx = this.clients.indexOf(port);
+    if (portIdx > -1) {
+      return;
+    }
```

Comment:
```suggestion
    if (!this.clients.includes(port)) return;
```

Reply:
Frigging ES6. Seriously it's a completely different language.